src/internal/docker/cli.go

```diff
@@ -184,25 +189,39 @@ func startContainer(ctx context.Context, containerName string, options RunOption
 	return run(ctx, "start", "-i", containerName)
 }
 
-func execContainer(ctx context.Context, containerName string, cmd []string, options RunOptions) error {
+func execContainer(ctx context.Context, yeyCtx yey.Context, containerName string, options RunOptions) error {
 	args := []string{"exec", "-ti"}
 	if options.WorkDir != "" {
 		args = append(args, "--workdir", options.WorkDir)
 	}
 	args = append(args, containerName)
-	args = append(args, cmd...)
+	args = append(args, yeyCtx.EntryPoint)
+	args = append(args, yeyCtx.Cmd...)
 
 	return run(ctx, args...)
 }
 
 func run(ctx context.Context, args ...string) error {
-	if yey.IsDryRun {
-		fmt.Printf("docker %s\n", strings.Join(args, " "))
-		return nil
+	if yey.IsDryRun || yey.IsVerbose {
+		cmd := fmt.Sprintf("docker %s", strings.Join(escapeArgs(args), " "))
+		if yey.IsDryRun {
+			fmt.Println(cmd)
+			return nil
+		}
+		yey.Log(cmd)
 	}
+
 	return attachStdPipes(exec.CommandContext(ctx, "docker", args...)).Run()
 }
 
+func escapeArgs(args []string) []string {
+	escaped := make([]string, 0, len(args))
+	for _, a := range args {
```

Comment:
what about other space characters? \n \t ?

Reply:
perhaps it might make more sense to quote the args?

Reply:
si on décide d'aller avec des quotes, il faudrait le faire uniquement lorsque c'est nécessaire, autrement, ça rendrait le dry-run vraiment illisible: tous les arguments seraient quotés, ça serait un spaghetti de double-quotes.  C'est pour ça que j'ai préféré y aller avec le escaping, qui est plus discret, moins intrusif.  Oui, il faudrait idéalement gérer les autres caractères spéciaux et ça serait peut-être plus simple avec le quoting, mais faudrait qu'on connaisse la liste exacte de ces caractères, qu'on teste pour leur présence et qu'on quote juste si on en trouve.

Reply:
you could match against the regexp `\s` and quote it if true